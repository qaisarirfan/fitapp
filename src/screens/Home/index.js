import React from "react"
import {View, TouchableOpacity} from "react-native"
import Entypo from "react-native-vector-icons/Entypo"
import PropTypes from "prop-types"
import {KeyboardAwareScrollView} from "react-native-keyboard-aware-scroll-view"

import HeaderBar from "../../components/HeaderBar"
import themeStyles from "./styles"

// Home Component content
export const Home = () => {
  const styles = themeStyles()

  return (
    <View style={styles.root} testID="welcome">
      <HeaderBar
        title="Home"
        isHeaderShow={false}
        rightContent={
          <TouchableOpacity testID="options_screen_button">
            <Entypo name="cog" size={32} color="#fff" />
          </TouchableOpacity>
        }
      />
      <KeyboardAwareScrollView
        testID="home_screen"
        scrollEnabled
        behavior="padding">
        <View style={styles.content} />
      </KeyboardAwareScrollView>
    </View>
  )
}

// Home Proptypes
Home.propTypes = {}

// Home Default props
Home.defaultProps = {}

export default Home
