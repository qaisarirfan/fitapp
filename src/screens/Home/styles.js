import {Dimensions, StyleSheet} from "react-native"

const screen = Dimensions.get("window")

// Home: Styling here
const styles = () => {
  return StyleSheet.create({
    root: {
      flex: 1,
      // backgroundColor: styleableTheme[500],
    },
    header: {
      alignItems: "flex-end",
      marginHorizontal: 20,
    },
    content: {
      paddingTop: screen.height * 0.04,
    },
    textHeader: {
      // color: styleableTheme[50],
      fontWeight: "bold",
      fontSize: 30,
      textAlign: "center",
      marginBottom: 20,
    },
  })
}

export default styles
