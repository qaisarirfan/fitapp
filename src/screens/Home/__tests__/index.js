import React from "react"
import {render, cleanup, store} from '../../../../jest/testUtils';
import Home from ".."

afterEach(() => {
  cleanup();
  store.clearActions();
});

describe("<Home />", () => {
  const createProps = () => ({
    navigation: {}
  })
  it("Does not explode", () => {
    const props = createProps()
    const rendered = render(<Home {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  })
})
