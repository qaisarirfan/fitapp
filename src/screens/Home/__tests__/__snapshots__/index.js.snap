// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Does not explode 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#4f6d7a",
      "flex": 1,
    }
  }
  testID="welcome"
>
  <View
    style={
      Array [
        Object {},
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 20,
          },
          Object {
            "backgroundColor": "#36505d",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "space-between",
            "paddingBottom": 12,
            "paddingTop": 12,
          },
          Object {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "justifyContent": "center",
            "paddingLeft": 6,
            "width": 250,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "justifyContent": "center",
            "paddingRight": 6,
            "width": 250,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="options_screen_button"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 32,
                },
                undefined,
                Object {
                  "fontFamily": "Entypo",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ï…œ
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    behavior="padding"
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    testID="home_screen"
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "paddingTop": 53.36,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginBottom": 40,
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "height": 300,
                "width": 300,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#eaedef",
              "fontSize": 30,
              "fontWeight": "bold",
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Currency Converter
        </Text>
        <View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#FFF",
                  "borderColor": "#a7b6bd",
                  "borderRadius": 5,
                  "borderWidth": 0.5,
                  "flexDirection": "row",
                  "marginHorizontal": 20,
                  "marginVertical": 10,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#FFF",
                  "borderBottomLeftRadius": 5,
                  "borderRightColor": "#cad3d7",
                  "borderRightWidth": 0.5,
                  "borderTopLeftRadius": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
              />
            </View>
            <TextInput
              allowFontScaling={true}
              keyboardType="numeric"
              onChangeText={[Function]}
              rejectResponderTermination={true}
              style={
                Object {
                  "color": "#00476E",
                  "flex": 1,
                  "fontSize": 16,
                  "padding": 10,
                }
              }
              testID="conversion_input_1"
              underlineColorAndroid="transparent"
              value="100"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#FFF",
                  "borderColor": "#a7b6bd",
                  "borderRadius": 5,
                  "borderWidth": 0.5,
                  "flexDirection": "row",
                  "marginHorizontal": 20,
                  "marginVertical": 10,
                },
                Object {
                  "backgroundColor": "#f0f0f0",
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#FFF",
                  "borderBottomLeftRadius": 5,
                  "borderRightColor": "#cad3d7",
                  "borderRightWidth": 0.5,
                  "borderTopLeftRadius": 5,
                  "opacity": 1,
                  "padding": 15,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
              />
            </View>
            <TextInput
              allowFontScaling={true}
              editable={false}
              rejectResponderTermination={true}
              style={
                Object {
                  "color": "#00476E",
                  "flex": 1,
                  "fontSize": 16,
                  "padding": 10,
                }
              }
              testID="conversion_input_2"
              underlineColorAndroid="transparent"
              value="0.00"
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginVertical": 20,
                "opacity": 1,
              }
            }
            testID="reverse_button"
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 20,
                  "marginRight": 10,
                  "width": 20,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 16,
                }
              }
            >
              Reverse Currencies
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "right": 0,
              },
              Object {
                "height": 0,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
